### ISSUE TITLE
nbdiff-web without an internet connection

### ISSUE THREAD

**Issue by mgunyho** [jupyter/nbdime#501]
I would like to use `nbdiff-web` on a lab computer without an internet connection. It works, but it takes around two minutes to load. The problem seems to be that it tries to fetch `font-awesome.min.css`, `promise-6.1.0.js` and `require.js` from the internet, see picture below from Chrome network analyzer.

![nbdiff-web-load-time](https://user-images.githubusercontent.com/20118130/67089105-8685d780-f1af-11e9-81ea-43c8e0dca0b2.png)

The page is empty until all three of those requests have failed, and it starts loading the diff only after that. For now I've worked around this by just commenting out those libraries from `nbdimepage.html`.

The nbdime version is 1.1.0, on Windows. I tried both Chrome and Firefox, both had the same issue (although admittedly they are not the latest versions).

**Comment by vidartf** [comment]
Hm I agree that this should ideally work (at least without blocking the load!).

Some ideas/investigation:
- Modify the [template](https://github.com/jupyter/nbdime/blob/a9faba1e3c2cc7858ba294f5b8b0ea81347f972e/nbdime/webapp/templates/nbdimepage.html#L7-L9) to mark these as [`async`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attr-async) if appropriate.
- We can probably remove the Promise polyfill now, as most modern browsers support it (and I think the current code uses other things that would fail if the browser is older than that).
- It is unclear to me if the requirejs code is strictly needed or not. I think it might be safe to remove it, but this should probably be tested extensively before being merged.
- The font-awsome icons or used rather sparsely (for the webapp, only in the mergetool for drang-and-drop deletion of outputs), so this can clearly be marked as async.

**Comment by vidartf** [comment]
@mgunyho Do you have an opportunity to help with any of these points? You should be able to modify the template directly in your install to test what will work or not.

**Comment by mgunyho** [comment]
Yeah, I could help with these. I've done a couple of diffs now with all three of the scripts commented out in the template and I haven't noticed any problems. Of course I haven't done any rigorous testing and haven't looked that closely if something is broken compared to the online version. Do you have something specific in mind that I should try?

**Comment by vidartf** [comment]
I would leave the font-awesome one in, but with the [`async` ](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attr-async) attribute, so that it doesn't block the other things. The promise polyfill can be removed. It might be that some cell outputs rely on `require` being present to correctly render (rich HTML/JS outputs e.g.), so I'm not entirely sure what to do with that one (maybe do this as a separate PR from the other two).

**Comment by mgunyho** [comment]
The font-awesome CSS is included using a `<link />` tag, so it doesn't support the async attribute and takes 17 seconds to time out.

The CSS could either be loaded asynchronously using JS as is suggested in [this SO answer](https://stackoverflow.com/questions/24020254/async-true-for-css-link-tag), or maybe the stylesheet could be bundled as a static file as part of the plugin.

**Comment by vidartf** [comment]
Thanks for this. I think adding font-awesome to the webapp bundle is probably the way to go, I can have a look at that if you'd like.

**Comment by vidartf** [comment]
Fix proposed in #517.
